android-core/src/main/java/com/mparticle/networking/MParticleBaseClientImpl.java

```diff
@@ -103,26 +103,34 @@ protected MPUrl getUrl(Endpoint endpoint, @Nullable String identityPath, boolean
         NetworkOptions networkOptions = mConfigManager.getNetworkOptions();
         DomainMapping domainMapping = networkOptions.getDomain(endpoint);
         String url = NetworkOptionsManager.getDefaultUrl(endpoint);
+        // Default domain to use for URL generation when domain mapping is specified
+        String defaultDomain = url ;
         boolean isDefaultUrl = true;
-        if (domainMapping != null && !MPUtility.isEmpty(domainMapping.getUrl()) && !forceDefaultUrl) {
+        if (domainMapping != null && !MPUtility.isEmpty(domainMapping.getUrl())) {
             String domainMappingUrl = domainMapping.getUrl();
             isDefaultUrl = url.equals(domainMappingUrl);
             url = domainMappingUrl;
         }
         Uri uri;
-        if (endpoint != Endpoint.CONFIG & isDefaultUrl) {
-            url = getPodUrl(url, mConfigManager.getPodPrefix(), mConfigManager.isDirectUrlRoutingEnabled());
+        if (endpoint != Endpoint.CONFIG) {
+            if (isDefaultUrl) {
```

Comment:
this means that the domain mapping is not set right?  I would add that as a comment here.  This code gets super complicated, so the more comments the better.